pygments/lexers/elpi.py

```diff
@@ -2,142 +2,172 @@
     pygments.lexers.elpi
     ~~~~~~~~~~~~~~~~~~~~
 
-    Lexer for the `Elpi <http://github.com/LPCIC/elpi>`_ programming language.
+    Lexer for the `Elpi <https://github.com/LPCIC/elpi>`_ programming language.
 
     :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
     :license: BSD, see LICENSE for details.
 """
 
 from pygments.lexer import RegexLexer, bygroups, include
-from pygments.token import \
-    Text, Comment, Operator, Keyword, Name, String, Number
+from pygments.token import Text, Comment, Operator, Keyword, Name, String, Number
+
+__all__ = ["ElpiLexer"]
 
-__all__ = ['ElpiLexer']
 
 class ElpiLexer(RegexLexer):
     """
-    Lexer for the `Elpi <http://github.com/LPCIC/elpi>`_ programming language.
+    Lexer for the `Elpi <https://github.com/LPCIC/elpi>`_ programming language.
 
-    .. versionadded::2.11
+    .. versionadded:: 2.11
     """
 
-    name = 'Elpi'
-    aliases = ['elpi']
-    filenames = ['*.elpi']
-    mimetypes = ['text/x-elpi']
-    
-    lcase_re = r"[a-z]"
-    ucase_re = r"[A-Z]"
+    name = "Elpi"
+    aliases = ["elpi"]
+    filenames = ["*.elpi"]
+    mimetypes = ["text/x-elpi"]
+
     digit_re = r"[0-9]"
     schar2_re = r"([+*^?/<>`'@#~=&!])"
-    schar_re = r"({}|-|\$|_)".format(schar2_re)
-    idchar_re = r"({}|{}|{}|{})".format(lcase_re,ucase_re,digit_re,schar_re)
-    idcharstarns_re = r"({}+|(?=\.[a-z])\.{}+)".format(idchar_re,idchar_re)
-    symbchar_re = r"({}|{}|{}|{}|:)".format(lcase_re, ucase_re, digit_re, schar_re)
-    constant_re = r"({}{}*|{}{}*|{}{}*|_{}+)".format(ucase_re, idchar_re, lcase_re, idcharstarns_re,schar2_re, symbchar_re,idchar_re)
-    symbol_re=r"(,|<=>|->|:-|;|\?-|->|&|=>|\bas\b|\buvar\b|<|=<|=|==|>=|>|\bi<|\bi=<|\bi>=|\bi>|\bis\b|\br<|\br=<|\br>=|\br>|\bs<|\bs=<|\bs>=|\bs>|@|::|\[\]|`->|`:|`:=|\^|-|\+|\bi-|\bi\+|r-|r\+|/|\*|\bdiv\b|\bi\*|\bmod\b|\br\*|~|\bi~|\br~)"
-    escape_re=r"\(({}|{})\)".format(constant_re,symbol_re)
-    const_sym_re = r"({}|{}|{})".format(constant_re,symbol_re,escape_re)
+    schar_re = rf"({schar2_re}|-|\$|_)"
+    idchar_re = rf"([a-zA-Z]|{digit_re}|{schar_re})"
+    symbchar_re = rf"({idchar_re}|:)"
+    constant_re = rf"([A-Z]{idchar_re}*|[a-z]{idchar_re}*(\.[a-zA-Z]{idchar_re}*)*|{schar2_re}{symbchar_re}*|_{idchar_re}+)"
```

Comment:
Note that this doesn't allow dots when the constant begins with a capital letter. I assume that is correct as it was already the case before.

Reply:
it is